lib/active_admin/views/components/paginated_collection.rb

```diff
@@ -102,8 +102,8 @@ def build_pagination
           # you pass in the :total_pages option. We issue a query to determine
           # if there is another page or not, but the limit/offset make this
           # query fast.
-          offset = collection.offset(collection.current_page * @per_page.to_i).limit(1).count
-          options[:total_pages] = collection.current_page + offset
+          another = collection.offset(collection.current_page * @per_page.to_i).any?
+          options[:total_pages] = collection.current_page + (another ? 1 : 0)
```

Comment:
This changes are new. Are they intentional?

Reply:
@deivid-rodriguez I see. Actually I made that change some days ago in my local branch to fix another issue that was trivial. 

Let's assume we have a `collection` that contains `ActiveRecord::QueryMethods#group` method by dint of which, it returns a relation of `hash` instead of array-convertible relation. The old code (`limit(1).count`) could only count the size if its a collection of array, but return a hash when the collection query contains `group_by` instead of count. For that, we would need `limit(1).count.count`, which is not pretty either.

With the old code, it would return an error on line#106 (`options[:total_pages] = collection.current_page + offset`) when the collection query contains `group_by`:
```
TypeError (Hash can't be coerced into Integer)
```

So I rethought the process and found the whole point is to determine if there is another page or not, as commented on line#103. So I used a safe solution `.any?` and added `1/0` based on `true/false` that **does the same thing as the old code** but will also be **able to handle collections that contains grouped result**.

---

As it seems to be a trivial change, I thought to push it along with this PR. But if you suggest, I will remove this change from this PR.

I am sorry if it makes issue in any way.

Thanks!

Reply:
I think this is a fix for https://github.com/activeadmin/activeadmin/issues/1123, which also has a PR for it already [opened](https://github.com/activeadmin/activeadmin/pull/5200), although your solution seems better to me at first sight?.

Let's remove it from this PR and handle it separately in another PR, ok?

Reply:
Feel free to submit your solution as a separate PR. I think the biggest issue preventing that PR (or yours) from being merged is that we'd like a spec for it!

Reply:
Ah I see! will keep in mind to include the spec for that in a new PR :+1: